astropy/coordinates/earth.py

```diff
@@ -645,33 +645,47 @@ def get_gcrs_posvel(self, obstime):
             The GCRS velocity of the object
         """
         # GCRS position
-        gcrs_data = self._get_gcrs(obstime).data
+        gcrs_data = self.get_gcrs(obstime).data
         obsgeopos = gcrs_data.without_differentials()
         obsgeovel = gcrs_data.differentials['s'].to_cartesian()
         return obsgeopos, obsgeovel
 
-    def _gravitational_redshift(self, obstime):
+    _GM_moon = consts.G * 7.34767309e22*u.kg
+    def gravitational_redshift(self, obstime,
+                               bodies=('sun', 'jupiter', 'moon', 'earth'),
```

Comment:
One problem with this approach is that the code later in the function *assumes* that Earth is the last body in this list, whilst the flexibility you've added can break this.

Reply:
Perhaps, since this is implicitly for GR at an EarthLocation, leave the Earth off the list and note in the docstring that it's always included?